src/validator/has-property.ts

```diff
@@ -0,0 +1,33 @@
+import {STRule} from '../rule/rule';
+import {STRuleFn} from '../rule/fn';
+import {STRuleModifiers} from '../rule/modifiers';
+import {STRuleNode} from '../rule/node';
+import {STRuleNodeType} from '../rule/node-type';
+
+export type STOpHasProperty<CallerType> = (propName: string) => CallerType;
+
+function hasProperty(obj: any, propName: string): boolean {
```

Comment:
Checking whether a property exists using property name directly works but can produce false positives or negatives. It's better to use `obj.hasOwnProperty(propertyName)` to be sure, even though it is technically slower than direct index access. It checks whether the propertyName exists _directly_ on the object.

Note: It is a function on obj, so the call will throw if obj does not exist. Drop out if obj is not present.

- [ ] Replace hasProperty contents with obj.hasOwnProperty